contracts/chain-adapters/Optimism_Adapter.sol

```diff
@@ -46,6 +46,7 @@ contract Optimism_Adapter is Base_Adapter, CrossDomainEnabled, Lockable {
         emit MessageRelayed(target, message);
     }
 
+    // todo: try making this delegate call
```

Comment:
trying this out locally. So I'll put up a PR for this if it add significant savings.